src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberTransitionProbabilityCacheCollection.java

```diff
@@ -0,0 +1,180 @@
+                : transitionMatrixCollection);
+        padded = padStates;
+        if (padded) {
+            maxCopyNumber = transitionMatrixCollection.getMaxCopyNumberInCollection();
+        } else {
+            maxCopyNumber = 0;
+        }
+        sexGenotypesSet = Collections.unmodifiableSet(transitionMatrixCollection.getSexGenotypes());
+        contigsSet = Collections.unmodifiableSet(transitionMatrixCollection.getContigs());
+    }
+
+    /**
+     * Public constructor from a transition matrix collection reader
+     *
+     * @param reader
+     * @param parentPath
+     * @param padStates
+     */
+    public IntegerCopyNumberTransitionProbabilityCacheCollection(@Nonnull final Reader reader,
+                                                                 @Nonnull final String parentPath,
+                                                                 final boolean padStates) {
+        final IntegerCopyNumberTransitionMatrixCollection transitionMatrixCollection =
+                IntegerCopyNumberTransitionMatrixCollection.read(reader, parentPath);
+        sexGenotypeAndContigToTransitionMatrixCacheMap = new HashMap<>();
+        constructCacheMapFromMatrixCollection(padStates ? transitionMatrixCollection.cloneWithPaddedStates()
+                : transitionMatrixCollection);
+        padded = padStates;
+        if (padded) {
+            maxCopyNumber = transitionMatrixCollection.getMaxCopyNumberInCollection();
+        } else {
+            maxCopyNumber = 0;
+        }
+        sexGenotypesSet = Collections.unmodifiableSet(transitionMatrixCollection.getSexGenotypes());
+        contigsSet = Collections.unmodifiableSet(transitionMatrixCollection.getContigs());
+    }
+
+    /**
+     * Public constructor from a transition matrix collection file
+     *
+     * @param inputFile
+     * @param padStates
+     */
+    public IntegerCopyNumberTransitionProbabilityCacheCollection(@Nonnull final File inputFile,
+                                                                 final boolean padStates) {
+        final IntegerCopyNumberTransitionMatrixCollection transitionMatrixCollection =
+                IntegerCopyNumberTransitionMatrixCollection.read(inputFile);
+        sexGenotypeAndContigToTransitionMatrixCacheMap = new HashMap<>();
+        constructCacheMapFromMatrixCollection(padStates ? transitionMatrixCollection.cloneWithPaddedStates()
+                : transitionMatrixCollection);
+        padded = padStates;
+        if (padded) {
+            maxCopyNumber = transitionMatrixCollection.getMaxCopyNumberInCollection();
+        } else {
+            maxCopyNumber = 0;
+        }
+        sexGenotypesSet = Collections.unmodifiableSet(transitionMatrixCollection.getSexGenotypes());
+        contigsSet = Collections.unmodifiableSet(transitionMatrixCollection.getContigs());
+    }
+
+    private void constructCacheMapFromMatrixCollection(
```

Comment:
Also, in cases like this I prefer to make static methods that return instances to be assigned to the fields.  That is, this could be a static method that returns a HashMap.  This prevents the possibility that one might inadvertently call such a non-static method more than once and leave the field in a bad state.